### PR TITLE
Update docs/Android.md

### PR DESCRIPTION
**Pull request by lanza**
Update docs/Android.md to reflect the changes in the NDK and https://github.com/apple/swift/pull/15833.

### CODE REVIEW

File: docs/Android.md [hunk 1]
```diff
@@ -67,144 +61,29 @@ The steps are as follows:
...
-
-```
-$ ./build.sh
-```
-
-Confirm that the build script created `armeabi-v7a/icu/source/i18n` and
-   `armeabi-v7a/icu/source/common` directories within your
-   `libiconv-libicu-android` directory.
+4. Change directories into `libiconv-libicu-android`: `cd libiconv-libicu-android`
+5. Run the Swift build script: `./build-swift.sh`
```
Comment (gonzalolarralde) [inline docs/Android.md]: Unless this changed in the libiconv-libicu-android repo, you'll need an env-var `LIBSUFFIX="swift"` for the files to be generated as the rest of the document assumes they're generated.

Comment (gonzalolarralde) [reply docs/Android.md]: Ah, ok, that makes sense. Ignore this then. Are those changes published somewhere?

Comment (modocache) [reply docs/Android.md]: Sorry for the wait @lanza, I granted you access to that repository.

**Comment by lanza** [comment]
@compnerd please smoke test and merge

**Comment by compnerd** [comment]
@swift-ci please smoke test and merge

**Pull request merged**